aicssegmentation/bin/batch_processing.py

```diff
@@ -252,70 +312,26 @@ def execute(self, args):
                 )
 
         elif args.mode == PER_DIR:
-
             filenames = glob(args.input_dir + "/*" + args.data_type)
             # [os.path.basename(os.path.splitext(f)[0])
             #             for f in os.listdir(args.input_dir)
             #             if f.endswith(args.data_type)]
             filenames.sort()
 
-            for _, fn in enumerate(filenames):
-
-                if os.path.exists(
-                    str(
-                        output_path
-                        / (
-                            os.path.splitext(os.path.basename(fn))[0]
-                            + "_struct_segmentation.tiff"
-                        )
-                    )
-                ):
-                    print(f"skipping {fn} ....")
-                    continue
-
-                image_reader = aicsimageio.AICSImage(fn)
-                img = image_reader.data
-                # import pdb; pdb.set_trace()
-
-                # fixing the image reading
-                if len(img.shape) == 6:
-                    # when z and c is not in order
-                    if img.shape[-3] < img.shape[-4]:
-                        img = np.transpose(img, (0, 1, 3, 2, 4, 5))
-                    struct_img = img[0, 0, args.struct_ch, :, :, :].astype(np.float32)
-                else:
-                    # when z and c is not in order
-                    if img.shape[-3] < img.shape[-4]:
-                        img = np.transpose(
-                            img,
-                            (
-                                0,
-                                2,
-                                1,
-                                3,
-                                4,
-                            ),
-                        )
-                    struct_img = img[0, args.struct_ch, :, :, :].astype(np.float32)
-
-                # Check if the segmenation is mitotic stage specific
-                if args.mitotic_stage is None:
-                    SegModule(
-                        struct_img,
-                        self.rescale_ratio,
-                        args.output_type,
-                        output_path,
-                        os.path.splitext(os.path.basename(fn))[0],
-                    )
-                else:
-                    SegModule(
-                        struct_img,
-                        args.mitotic_stage,
-                        self.rescale_ratio,
-                        args.output_type,
-                        output_path,
-                        fn,
-                    )
+            if args.dask:
+                # using dask offers ~6x speedup on the template segmentation.
+                import dask
+
+                lazy_results = []
```

Comment:
Just for clarification, what is the purpose of "lazy_results" and what is being stored in there?

Reply:
My understanding is that it holds the setup for all the calculations, and then when you call dask.compute it does the actual computation that's been set up. 

Here's a link to the dask parallelization tutorial: https://examples.dask.org/applications/embarrassingly-parallel.html